core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -1,105 +1,689 @@
 package org.evomaster.core.problem.graphql
 
-import ch.qos.logback.classic.db.names.TableName
 import com.google.gson.Gson
-import io.swagger.v3.oas.models.OpenAPI
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.problem.graphql.param.GQInputParam
+import org.evomaster.core.problem.graphql.param.GQReturnParam
 import org.evomaster.core.problem.graphql.schema.SchemaObj
+import org.evomaster.core.problem.graphql.schema.__TypeKind
+import org.evomaster.core.problem.rest.param.Param
 import org.evomaster.core.search.Action
+import org.evomaster.core.search.gene.*
 import java.util.concurrent.atomic.AtomicInteger
-
+import java.util.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
 object GraphQLActionBuilder {
 
+    private val log: Logger = LoggerFactory.getLogger(GraphQLActionBuilder::class.java)
     private val idGenerator = AtomicInteger()
+    var tables: MutableList<Table> = mutableListOf()
+    var argsTables: MutableList<Table> = mutableListOf()
+    var tempArgsTables: MutableList<Table> = mutableListOf()
 
     /**
      * @param schema: the schema extracted from a GraphQL API, as a JSON string
      * @param actionCluster: for each mutation/query in the schema, populate this map with
      *                      new action templates.
      */
+
     fun addActionsFromSchema(schema: String, actionCluster: MutableMap<String, Action>) {
 
         val gson = Gson()
         val schemaObj: SchemaObj = gson.fromJson(schema, SchemaObj::class.java)
 
         /*
          TODO
-
             - go through every Query and Mutation
-            - create action for it which the needed genes// asma: Not yet
+            - create action for it which the needed genes
             - add the action to actionCluster
          */
 
-        var table : MutableList<Table> = mutableListOf()
 
+        initTablesInfo(schemaObj)
+        for (element in tables) {
+            if (element.tableType == "Mutation" || element.tableType == "Query") {
+                handleOperation(actionCluster,
+                        element.tableField,
+                        element.tableType,
+                        element.tableFieldType,
+                        element.kindOfTableFieldType.toString(),
+                        element.kindOfTableField.toString(),
+                        element.tableType.toString(),
+                        element.isKindOfTableFieldTypeOptional,
+                        element.isKindOfTableFieldOptional,
+                        element.tableFieldWithArgs
+                )
+            }
+        }
 
-        for (elementIntypes in schemaObj.data?.__schema?.types.orEmpty()) {
+    }
 
-            if (elementIntypes.name == "__Schema") {
+    private fun initTablesInfo(schemaObj: SchemaObj) {
```

Comment:
this is a 300+ LOCs method, with quite a bit of duplication. needs to be refactored/simplified. we can do it together